src/main/java/io/github/slsa/models/Recipe.java

```diff
@@ -0,0 +1,91 @@
+package io.github.slsa.models;
+
+import java.util.Objects;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Identifies the configuration used for the build. When combined with materials, this SHOULD fully
+ * describe the build, such that re-running this recipe results in bit-for-bit identical output (if
+ * the build is reproducible).
+ *
+ * <p>MAY be unset/null if unknown, but this is DISCOURAGED.
+ *
+ * <p>NOTE: The Recipe entity has additional properties: arguments and environment that are
+ * classified as generic objects in the spec. For this reason it would be expected that builders
+ * using this library would extend from the default Recipe and create their own custom class.
+ */
+public class Recipe {
+
+  /**
+   * URI indicating what type of recipe was performed. It determines the meaning of
+   * recipe.entryPoint, recipe.arguments, recipe.environment, and materials. (<a
+   * href="https://github.com/in-toto/attestation/blob/main/spec/field_types.md#TypeURI">TypeURI</a>)
+   */
+  @NotNull(message = "recipe type must not be blank")
+  private String type;
+
+  /**
+   * Index in materials containing the recipe steps that are not implied by recipe.type. For
+   * example, if the recipe type were “make”, then this would point to the source containing the
+   * Makefile, not the make program itself.
+   *
+   * <p>Omit this field (or use null) if the recipe does’t come from a material.
```

Comment:
Can an 'int' be omitted or set to null?

Maybe https://docs.oracle.com/javase/8/docs/api/java/util/OptionalInt.html ?